File: src/main/java/com/jme/asset/store/entity/UserEntity.java
package com.jme.asset.store.entity;

import com.jme.asset.store.entity.BaseEntity;
import org.apache.catalina.LifecycleState;

import javax.persistence.*;
import java.util.List;

/**
 * The user entity.
 *
 * @author Denis Lesheniuk
 */
@Entity
@Table(name = "USER_ENTITY",
        indexes = {
                @Index(columnList = "login", name = "user_login_index")
        })
public class UserEntity extends BaseEntity {

    /**
     * The user login.
     */
    @Column(name = "login", length = 50, nullable = false, unique = true)
    private String login;

    /**
     * The user first name.
     */
    @Column(name = "first_name", length = 50)
    private String firstName;

    /**
     * The user last name.
     */
    @Column(name = "last_name", length = 50)
    private String lastName;

    /**
     * The user middle name.
     */
    @Column(name = "middle_name", length = 50)
    private String middleName;

    /**
     * The user mail.
     */
    @Column(name = "mail", length = 50, unique = true)
    private String mail;

    /**
     * The user password.
     */
    @Column(name = "password", length = 50, nullable = false)
    private String password;

    /**
     * The user roles.
     */
    @ManyToMany(fetch = FetchType.EAGER)
    private List<RoleEntity> roles;

    /**
     * Get the user login
     *
     * @return the user login.
     */
    public String getLogin() {
        return login;
    }

    /**
     * Set the user login
     *
     * @param login the user login.
     */
    public void setLogin(String login) {
        this.login = login;
    }

    /**
     * Get the user first name
     *
     * @return the user first name.
     */
    public String getFirstName() {
        return firstName;
    }

    /**
     * Set the user first name
     *
     * @param firstName the user first name
     */
    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    /**
     * Get the user last name
     *
     * @return the user last name.
     */
    public String getLastName() {
        return lastName;
    }

    /**
     * Set the user last name
     *
     * @param lastName the user last name.
     */
    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    /**
     * Get the user middle name
     *
     * @return the user middle name.
     */
    public String getMiddleName() {
        return middleName;
    }

    /**
     * Set the user middle name
     *
     * @param middleName the user middle name.
     */
    public void setMiddleName(String middleName) {
        this.middleName = middleName;
    }

    /**
     * Get the user e-mail
     *
     * @return the user mail.
     */
    public String getMail() {
        return mail;
    }

    /**
     * Get set the user e-mail
     *
     * @param mail the user e-mail.
     */
    public void setMail(String mail) {
        this.mail = mail;
    }

    /**
     * Get the user password
     *
     * @return the user password.
     */
    public String getPassword() {
        return password;
    }

    /**
     * Set the user password
     *
     * @param password the user password.
     */
    public void setPassword(String password) {
        this.password = password;
    }

    /**
     * Get the user roles
     *
     * @return the user roles.
     */
    public List<RoleEntity> getRoles() {
        return roles;
    }

    /**
     * Set the user roles
     *
     * @param roles user roles.
     */
    public void setRoles(List<RoleEntity> roles) {
        this.roles = roles;
    }

    @Override
    public String toString() {
        return "UserEntity{" +
                "login='" + login + '\'' +
                ", firstName='" + firstName + '\'' +
                ", lastName='" + lastName + '\'' +
                ", middleName='" + middleName + '\'' +
                ", mail='" + mail + '\'' +
                ", password='" + password + '\'' +
                ", roles=" + roles +
                '}';
    }

    /**
     * Add role to the user
     *
     * @param role the role name
     */
    public void addRole(final RoleEntity role) {
        if (roles.contains(role)) {
            throw new RuntimeException("role is already exists");
        }
        roles.add(role);
    }

    /**
     * Remove role from the user
     *
     * @param role the role name
     */
    public void removeRole(final RoleEntity role) {
        if (roles.contains(role)) {
            throw new RuntimeException("role is already exists");
        }
        roles.remove(role);
    }
}

File: front-end/src/model/entity/user-entity.ts
export class UserEntity{

  private _login: string;

  private _first_name: string;

  private _last_name: string;

  private _middle_name: string;

  private _mail: string;

  private _password: string;

  private _roles: string[];

  constructor(login: string, first_name: string, last_name: string, middle_name: string, mail: string, password: string, roles: string[]) {
    this._login = login;
    this._first_name = first_name;
    this._last_name = last_name;
    this._middle_name = middle_name;
    this._mail = mail;
    this._password = password;
    this._roles = roles;
  }


  get login(): string {
    return this._login;
  }

  set login(value: string) {
    this._login = value;
  }

  get first_name(): string {
    return this._first_name;
  }

  set first_name(value: string) {
    this._first_name = value;
  }

  get last_name(): string {
    return this._last_name;
  }

  set last_name(value: string) {
    this._last_name = value;
  }

  get middle_name(): string {
    return this._middle_name;
  }

  set middle_name(value: string) {
    this._middle_name = value;
  }

  get mail(): string {
    return this._mail;
  }

  set mail(value: string) {
    this._mail = value;
  }

  get password(): string {
    return this._password;
  }

  set password(value: string) {
    this._password = value;
  }

  get roles(): string[] {
    return this._roles;
  }

  set roles(value: string[]) {
    this._roles = value;
  }
}
